src/ChannelServer/Network/ChannelPacketSender.cs

```diff
@@ -1472,32 +1472,57 @@ public static void ZC_MOVE_DIR(Character character, float x, float y, float z, f
 			packet.PutFloat(dx);
 			packet.PutFloat(dy);
 			packet.PutByte(1); // 0 = reduced movement speed... walk mode?
-			packet.PutFloat(character.GetSpeed());
+            packet.PutFloat(character.Speed);
 			packet.PutByte(1);
 			packet.PutFloat(unkFloat);
 
-			character.Map.Broadcast(packet, character);
-		}
-
-		/// <summary>
-		/// Broadcasts ZC_MOVE_STOP in range of character, informing other
-		/// characters about the movement stop.
-		/// </summary>
-		/// <param name="character"></param>
-		/// <param name="x"></param>
-		/// <param name="y"></param>
-		/// <param name="z"></param>
-		public static void ZC_MOVE_STOP(Character character, float x, float y, float z)
+            character.Map.Broadcast(packet, character);
+		}
+
+        /// <summary>
+        /// Broadcasts ZC_MOVE_PATH in range of monster, informing
+        /// characters about the movement.
+        /// </summary>
+        /// <param name="monster"></param>
+        /// <param name="x"></param>
+        /// <param name="y"></param>
+        /// <param name="z"></param>
+        public static void ZC_MOVE_PATH(Monster monster, float x, float y, float z)
+        {
+            var packet = new Packet(Op.ZC_MOVE_PATH);
+
+            packet.PutInt(monster.Handle);
+            packet.PutFloat(monster.Position.X);
```

Comment:
ZC_MOVE_PATH contains move vector with coordinates described by int variables, not float. Or am i wrong? Note: Also it contains 2 unknown float after path (first seems to be speed, and second always zero)
Prove:
http://i.imgur.com/iGkvGYD.png
http://mabinogi.pp.ua/g/8d127679d00830f32385bceca3229616.png